workspaces/diff-engine/lib/index.js

```diff
@@ -16,6 +16,51 @@ function spawn({ specPath }) {
   const output = new PassThrough();
   const error = new PassThrough();
 
+  const binPath = getBinPath();
+
+  const diffProcess = Execa(binPath, [specPath], {
+    input,
+    stdio: 'pipe',
+  });
+
+  if (!diffProcess.stdout || !diffProcess.stderr)
+    throw new Error('diff process should have stdout and stderr streams');
+
+  diffProcess.stdout.pipe(output);
+  diffProcess.stderr.pipe(error);
+
+  // get a clean result promise, so we stay in control of the exact API we're exposing
+  const result = diffProcess.then(
+    (childResult) => {},
+    (childResult) => {
+      throw new DiffEngineError(childResult);
+    }
+  );
+  return { input, output, error, result };
+}
+
+function readSpec({ specDirPath }) {
+  const output = new PassThrough();
+
+  const binPath = getBinPath();
+
+  const assembleProcess = Execa(binPath, [specDirPath, 'assemble'], {
+    stdio: ['ignore', 'pipe', 'inherit'],
+  });
+
+  assembleProcess.stdout.pipe(output);
+
+  // surface any errors in execution on the output stream
+  assembleProcess.then(
+    (childResult) => {},
+    (childResult) => {
+      output.emit('error', new DiffEngineError(childResult));
+    }
+  );
+  return output;
+}
```

Comment:
This is the addition to diff-engine's thin Node.js layer, allowing consuming code to call `readSpec({ specDirPath })` without having to worry about spawning binaries. Made sure to get the `stdio` right this time, with the results the only thing we want back-pressure and `stderr` being forwarded to the parent process. That simplified the return value a lot to, with a single stream being enough.